libtenzir/include/tenzir/plugin.hpp

```diff
@@ -798,6 +798,9 @@ class context {
   /// Inspects the context.
   virtual auto show() const -> record = 0;
 
+  /// Dumps the context content or a dumping error.
+  virtual auto dump() -> caf::expected<std::vector<table_slice>> = 0;
```

Comment:
How does this "stream" events to the `context inspect` operator when it returns a single vector as opposed to a generator?

Reply:
The generator right now is inside the context implementation. Launching `dump()` advances that generator and returns the yielded element in a vector.

This would probably be way easier (and more correct) if I could just straight up return a `generator<table_slice>` to the operator. But how would I make a `generator` inspectable?

Reply:
You can't transfer the generator to the operator, but you can at least return a generator from the context and hold the instantiated coroutines in the context actor instead of the context itself.

Reply:
That is what I have been actually working on right now. Seems like a good compromise.

Reply:
FWIW, if we ever manage to actually upgrade to CAF 0.19 then we can transfer generators between actors, but unfortunately it's not possible until then. Once the database plugin is done we can start migrating to the newest version of CAF because then we'll finally have gotten rid of all the places where we currently use CAF streaming.

Reply:
I made use of storing `generator`s in the `context_actor`.